eng/CodeGeneration.targets

```diff
@@ -0,0 +1,26 @@
+<Project>
+
+  <PropertyGroup>
+    <_AutoRestVersion>https://github.com/Azure/autorest/releases/download/autorest-3.0.6173/autorest-3.0.6173.tgz</_AutoRestVersion>
+    <_AutoRestCoreVersion>3.0.6212</_AutoRestCoreVersion>
+    <_AutoRestCSharpVersion>https://github.com/Azure/autorest.csharp/releases/download/1.0.0-dev.20200210.1/autorest-csharp-v3-1.0.0-dev.20200210.1.tgz</_AutoRestCSharpVersion>
+    <_SupportsCodeGeneration Condition="'$(IsClientLibrary)' == 'true'">true</_SupportsCodeGeneration>
+    <AutoRestConfiguration>autorest.md</AutoRestConfiguration>
+
+    <_SharedCodeDirectory>$(MSBuildThisFileDirectory)../sdk/core/Azure.Core/src/Shared/</_SharedCodeDirectory>
+    <_AutoRestSharedCodeDirectory>$(_SharedCodeDirectory)Autorest/</_AutoRestSharedCodeDirectory>
+
+    <_GenerateCode Condition="'$(_SupportsCodeGeneration)' == 'true' AND Exists('$(AutoRestConfiguration)')">true</_GenerateCode>
+    <UsesJsonSerialization Condition="'$(UsesJsonSerialization)' == ''">true</UsesJsonSerialization>
+  </PropertyGroup>
+
+  <Target Name="GenerateCode" Condition="'$(_GenerateCode)' == 'true'" >
+    <RemoveDir Directories="$(MSBuildProjectDirectory)/Generated"/>
+    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) --input=$(AutoRestConfiguration) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory) --verbose" />
```

Comment:
What is `npx` and are sure it on people's path?

Reply:
It's a tool that comes with npm, it allows executing a package without having it installed into node_modules. For now people using codegen would have to have node as a prerequisite.

Reply:
OK if it comes with the node install its fine. We should at least add some pre-reqs in the docs somewhere so folks know they need it to do the code generation.